src/eth_provider/database/types/transaction.rs

```diff
@@ -24,57 +25,37 @@ impl From<Transaction> for StoredTransaction {
+    pub fn arbitrary_with_optional_fields(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let transaction = Self::arbitrary(u)?.tx;
 
-        let transaction_type = Into::<u8>::into(transaction.tx_type()) % 3;
+        let transaction_type = Into::<u8>::into(transaction.transaction_type.unwrap_or_default()) % 3;
 
-        Ok(StoredTransaction {
+        Ok(Self {
             tx: Transaction {
-                hash: transaction.hash,
-                nonce: transaction.nonce(),
                 block_hash: Some(B256::arbitrary(u)?),
                 block_number: Some(u64::arbitrary(u)?),
                 transaction_index: Some(u64::arbitrary(u)?),
-                from: Address::arbitrary(u)?,
-                to: transaction.to(),
-                value: transaction.value(),
                 gas_price: Some(u128::arbitrary(u)?),
                 gas: u64::arbitrary(u)? as u128,
                 max_fee_per_gas: if TryInto::<TxType>::try_into(transaction_type).unwrap() == TxType::Legacy {
                     None
                 } else {
-                    Some(transaction.max_fee_per_gas())
+                    Some(u128::arbitrary(u)?)
                 },
                 max_priority_fee_per_gas: if TryInto::<TxType>::try_into(transaction_type).unwrap() == TxType::Legacy {
                     None
                 } else {
-                    Some(transaction.max_priority_fee_per_gas().unwrap_or_default())
+                    Some(u128::arbitrary(u)?)
                 },
-                max_fee_per_blob_gas: transaction.max_fee_per_blob_gas(),
-                input: transaction.input().clone(),
                 signature: Some(reth_rpc_types::Signature {
-                    r: transaction.signature.r,
-                    s: transaction.signature.s,
-                    v: U256::arbitrary(u)?,
                     y_parity: Some(reth_rpc_types::Parity(bool::arbitrary(u)?)),
+                    ..reth_rpc_types::Signature::arbitrary(u)?
                 }),
-                chain_id: transaction.chain_id(),
-                blob_versioned_hashes: transaction.blob_versioned_hashes(),
-                access_list: transaction.access_list().map(|list| {
-                    reth_rpc_types::AccessList(
-                        list.0
-                            .iter()
-                            .map(|item| reth_rpc_types::AccessListItem {
-                                address: item.address,
-                                storage_keys: item.storage_keys.clone(),
-                            })
-                            .collect(),
-                    )
-                }),
-                transaction_type: Some(Into::<u8>::into(transaction.tx_type()) % 3),
+                transaction_type: Some(transaction_type),
                 other: Default::default(),
+                access_list: Some(reth_rpc_types::AccessList::arbitrary(u)?),
+                ..transaction
```

Comment:
I guess same question: why do we have to use `access_list: Some(reth_rpc_types::AccessList::arbitrary(u)?),` for example, doesn't this get filled in with `..transaction`?

Reply:
Because it can be a `None`, raising this conversion error during conversion to primitive: https://github.com/paradigmxyz/reth/blob/57d09e84beaa4dafa505461db733d672163e4d99/crates/primitives/src/transaction/mod.rs#L682